Apps/Encounter/src/Delegates/RestMSPVisitDelegate.cs

```diff
@@ -116,10 +116,10 @@ public async Task<RequestResult<MSPVisitHistoryResponse>> GetMSPVisitHistoryAsyn
                     string payload = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
                     if (response.IsSuccessStatusCode)
                     {
-                        MSPVisitHistory visitHistory = JsonSerializer.Deserialize<MSPVisitHistory>(payload, options);
+                        MSPVisitHistory? visitHistory = JsonSerializer.Deserialize<MSPVisitHistory>(payload, options);
                         retVal.ResultStatus = Common.Constants.ResultType.Success;
-                        retVal.ResourcePayload = visitHistory.Response;
-                        retVal.TotalResultCount = visitHistory.Response?.TotalRecords;
+                        retVal.ResourcePayload = visitHistory?.Response;
```

Comment:
nit: instead of using the ? operator we could just check for the nullable part earlier and return an error if null